From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Tue, 20 Oct 2026 16:22:30 +0200
Subject: [PATCH] Allow the plugin id to be set at build time

Packagers that load the plugin from a different front end sometimes
need it registered under another protocol id than
"prpl-eionrobb-discord", for example to keep existing accounts working.
Guard the default DISCORD_PLUGIN_ID with #ifndef so it can be overridden
with -DDISCORD_PLUGIN_ID='"..."' instead of patching the source.

The default id is unchanged.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 libdiscord.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libdiscord.c b/libdiscord.c
index 31fc76c..468e34e 100644
--- a/libdiscord.c
+++ b/libdiscord.c
@@ -58,7 +58,9 @@
 #define purple_ssl_read(a, b, c)  ((a) && (a)->private_data ? purple_ssl_read((a), (b), (c)) : 0)
 
 #ifndef DISCORD_PLUGIN_OVERRIDE
+#ifndef DISCORD_PLUGIN_ID
 #define DISCORD_PLUGIN_ID "prpl-eionrobb-discord"
+#endif
 #define DISCORD_PLUGIN_NAME "Discord"
 #ifndef DISCORD_PLUGIN_VERSION
 #define DISCORD_PLUGIN_VERSION "1.0"
-- 
2.53.0

